### PR TITLE
MOD-5911: Report timeout in shards for `FT.SEARCH` in the coordinator

### PR DESCRIPTION
**Pull request by raz-mon**
Currently, the coordinator doesn't report timeouts experienced in the shards. This PR fixes this behavior, so that any timeouts experienced in the shards are propagated to the client.

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: tests/pytests/test_vecsim.py [hunk 1]
```diff
@@ -1678,7 +1678,7 @@ def test_timeout_reached():
                 res = conn.execute_command('FT.SEARCH', 'idx', '*=>[KNN $K @vector $vec_param]', 'NOCONTENT', 'LIMIT', 0, n_vec,
                                            'PARAMS', 4, 'K', n_vec, 'vec_param', query_vec.tobytes(),
                                            'TIMEOUT', 1)
-                env.assertEqual(res[0], timeout_expected)
+                env.assertEqual(str(res[0]), timeout_expected)
```
Comment (oshadmi) [inline tests/pytests/test_vecsim.py:1681]: Make sure the type is expected

**Review by oshadmi — DISMISSED** [review]
Followup is to bring back #3987

**Comment by raz-mon** [comment]
The replacement for #3987 is #4062.

**Review by oshadmi — APPROVED** [review]

**Pull request merged**